Lib/encodings/idna.py

```diff
@@ -121,7 +121,7 @@ def ToUnicode(label):
         except UnicodeError:
             raise UnicodeError("Invalid character in IDN label")
     # Step 3: Check for ACE prefix
-    if not label.startswith(ace_prefix):
+    if not label.lower().startswith(ace_prefix):
```

Comment:
The change is memory and CPU expensive; explicit check for first 3 characters could be much better I believe.